### ISSUE TITLE
pip does not install typing_extensions

### ISSUE THREAD

**Issue by j0ma** [bltlab/seqscore#3]
It seems like `pip` doesn't install `typing_extensions` when installing in an empty Conda environment that uses Python 3.8. After manually `pip install`ing it, though, `seqscore` seems to run fine. 

Log (output edited slightly to remove clutter):

```
> jonne@lignos07:~/softgaz/soft-gazetteers/outputs$ seqscore
seqscore: command not found
> jonne@lignos07:~/softgaz/soft-gazetteers/outputs$ pip install seqscore 2>&1 > /tmp/install_seqscore_pip.log
> jonne@lignos07:~/softgaz/soft-gazetteers/outputs$ seqscore --help
Traceback (most recent call last):
  File "/home/jonne/miniconda3/envs/test-pip-seqscore/bin/seqscore", line 5, in <module>
    from seqscore.scripts.seqscore import cli
  File "/home/jonne/miniconda3/envs/test-pip-seqscore/lib/python3.8/site-packages/seqscore/scripts/seqscore.py", line 7, in <module>
    from seqscore.conll import (
  File "/home/jonne/miniconda3/envs/test-pip-seqscore/lib/python3.8/site-packages/seqscore/conll.py", line 10, in <module>
    from seqscore.encoding import Encoding, EncodingError, get_encoding
  File "/home/jonne/miniconda3/envs/test-pip-seqscore/lib/python3.8/site-packages/seqscore/encoding.py", line 5, in <module>
    from typing_extensions import Protocol
ModuleNotFoundError: No module named 'typing_extensions'
> jonne@lignos07:~/softgaz/soft-gazetteers/outputs$ pip install typing_extensions 2>&1 > /dev/null
> jonne@lignos07:~/softgaz/soft-gazetteers/outputs$ seqscore --help
Usage: seqscore [OPTIONS] COMMAND [ARGS]...

Options:
  --version  Show the version and exit.
  --help     Show this message and exit.

Commands:
  convert
  count
  repair
  score
  validate

```

**Comment by ConstantineLignos** [comment]
I think we need two fixes:

1. The import should condition on whether we're less than Python 3.8, if so do `from typing_extensions import Protocol`, if not, `from typing import Protocol`: https://stackoverflow.com/a/1875584/944164
2. Conditionally install `typing extensions` in setup.py if Python is less than 3.8: https://setuptools.pypa.io/en/latest/userguide/dependency_management.html#platform-specific-dependencies

There's a third issue that I'd like to ignore for the moment, which is that our CI doesn't do any package testing before installing the test dependencies, and it appears the test dependencies install 3.7, which is why we never noticed a test failure on 3.7.

**Comment by ConstantineLignos** [comment]
Fixed by commits ab8d545 and 34f5ea4, thank you for contributing the fixes!
